controllers/mock.go

```diff
@@ -54,34 +54,45 @@ func (i ModelInterfaceMock) NewUser(
 
 // DBFunctionInterfaceMock overwrites the mysqldb package function implementations with mock code.
 type DBFunctionInterfaceMock struct {
-	user         *models.User
-	product      *models.Product
-	privileges   models.Privileges
-	userProducts *models.UserProducts
-	err          error
+	user                 *models.User
+	userDeleted          *bool
+	userAdded            *bool
+	product              *models.Product
+	productAdded         *bool
+	productDeleted       *bool
+	usersProductsUpdated *bool
+	privileges           models.Privileges
+	userProducts         *models.UserProducts
+	err                  error
 }
 
 func (i DBFunctionInterfaceMock) GetPrivileges() (models.Privileges, error) {
 	return i.privileges, i.err
 }
 
-func (i DBFunctionInterfaceMock) GetUserByEmail(email string, tx *sql.Tx) (*models.User, error) {
+func (i DBFunctionInterfaceMock) GetUser(queryString string, keyValue interface{}, tx *sql.Tx) (*models.User, error) {
 	return i.user, i.err
 }
 
 func (i DBFunctionInterfaceMock) AddUser(user *models.User, tx *sql.Tx) error {
+	*i.userAdded = true
```

Comment:
Why don't you use the pointer as the target of the method?
`func (i *DBFunctionInterfaceMock) AddUser(user *models.User, tx *sql.Tx) error { ... }`

This stands for the other methods too.

Reply:
Can you explain more why is pointer needed? I don't have knowledge in the area.

Reply:
See for more info: https://medium.com/globant/go-method-receiver-pointer-vs-value-ffc5ab7acdb

Reply:
I see. Now I understand why my values never changed when I tried to update something.... :+1: